p2p/service/service.go

```diff
@@ -14,7 +14,7 @@ type Message interface {
 // Service is an interface that represents a networking service (ideally p2p) that we can use to send messages or listen to incoming messages
 type Service interface {
 	Start() error
-	RegisterProtocol(protocol string) chan Message
+	RegisterProtocol(protocol string, bufferSize int) chan Message
```

Comment:
should receive the channel

Reply:
I didn't get that, *pass the chan into RegisterProtocal()*

Did you mean like changing RegisterProtocol signature to accept chan as parameter? 
Something like `RegisterProtocol(name, chan Message)`

Reply:
yes thats right

Reply:
@adigunhammedolalekan hi, nice work with the pr but you left this part out
can you fix this so we can merge ?

Reply:
Yes! I'll really love to, but can you please provide a little bit more details? I'll really appreciate that.

Reply:
> I didn't get that, _pass the chan into RegisterProtocol()_
> 
> Did you mean like changing RegisterProtocol signature to accept chan as parameter?
> Something like `RegisterProtocol(name, chan Message)`

you need to change  RegisterProtocol like we discussed and refactor the places it is used 
so the tests still pass
so basically the only thing to do is change  RegisterProtocol(protocol string, bufferSize int) chan Message to RegisterProtocol(protocol string, chan Message)